src/base/bittorrent/torrentinfo.cpp

```diff
@@ -308,9 +308,29 @@ int BitTorrent::TorrentInfo::fileIndex(const QString& fileName) const
     return -1;
 }
 
+bool TorrentInfo::hasRootFolder() const
+{
+    QString testRootFolder;
+    for (int i = 0; i < filesCount(); ++i) {
+        const QString filePath = this->filePath(i);
+        if (QDir::isAbsolutePath(filePath)) continue;
+
+        const QStringList filePathElements = filePath.split('/');
+        // if at least one file has no root folder there is no common root folder exists
```

Comment:
I meant to change the wording, one last try:
if one or more file has no root folder then there exist no common root folder
make sense now? if no, then nevermind...